cranelift/filetests/filetests/egraph/skeleton.clif

```diff
@@ -110,6 +125,21 @@ block0:
 ;     return v11  ; v11 = -1
 ; }
 
+function %cprop_srem() -> i32 {
+block0:
+    v0 = iconst.i32 -17
+    v1 = iconst.i32 7
+    v2 = srem v0, v1
+    return v2
+}
+
+; function %cprop_srem() -> i32 fast {
+; block0:
+;     v28 = iconst.i32 -3
+;     v2 -> v28
+;     return v28  ; v28 = -3
+; }
+
```

Comment:
Can we add tests for:

- `INT_MIN % -1`, for narrow and wide types (i8, i16, i32, i64)
- negative and positive values in numerator and denominator for narrow and wide types

and runtests that are known to trigger these optimizations as well? (For the runtests, set `opt_level`; we won't be able to directly assert on the compilation output, but we should at least lock down that the computations produce the correct value)